pkg/commands/ls.go

```diff
@@ -113,7 +113,14 @@ func mergeStagingComponents(deployed []apis.ComponentMeta, env *types.EnvMeta, i
 					v.Traits[j] = t
 				}
 				aspec.Components[i] = v
+				/* Before any traits is attached to the workload, if ApplicationConfiguration is retrieved through
+				OpenAPI, the .spec.Components[].Traits is nil; if it's resembled by local file, .spec.Components[].Traits
+				is the []v1alpha2.ComponentTrait, 0). Let's set it to be equal before comparing the whole .Spec field.*/
+				if compMeta.AppConfig.Spec.Components[i].Traits == nil && len(v.Traits) == 0 {
+					compMeta.AppConfig.Spec.Components[i].Traits = aspec.Components[i].Traits
+				}
```

Comment:
I am afraid the hack way here might not be the best solution. One solution is to change the response of client-go from `ApplicationConfiguration` to make `.spec.Components[].Traits` to be in Golang slice, the other way is to change the type of traits of https://github.com/oam-dev/kubevela/blob/master/pkg/appfile/service.go#L114 which I think the type is reasonable. 
```
traits := make([]v1alpha2.ComponentTrait, 0)
```
I could not find a way out of the dilemma, so here is a solution to the issue #534